Data/Scripts/001_Technical/002_RubyUtilities.rb

```diff
@@ -162,6 +159,145 @@ def self.move(source, destination)
   end
 end
 
+#===============================================================================
+#  Extensions for the `Color` class
+#===============================================================================
+class Color
+  # alias for old constructor
+  alias init_original initialize unless self.private_method_defined?(:init_original)
+  #-----------------------------------------------------------------------------
+  # new constructor accepts RGB values as well as a hex number or string value
+  #-----------------------------------------------------------------------------
+  def initialize(*args)
+  	pbPrintException("Wrong number of arguments! At least 1 is needed!") if args.length < 1
+  	if args.length == 1
+      if args[0].is_a?(Fixnum)
+        hex = args[0].dup.to_s(16)
+      elsif args[0].is_a?(String)
+        hex = args[0].dup
+        hex.gsub!("#", "") if hex.include?("#")
+      end
+      pbPrintException("Wrong type of argument given!") if !hex
+      r = hex[0...2].to_i(16)
+      g = hex[2...4].to_i(16)
+      b = hex[4...6].to_i(16)
+  	elsif args.length == 3
+      r, g, b = *args
+  	end
+  	return init_original(r, g, b) if r && g && b
+  	return init_original(*args)
+  end
+  #-----------------------------------------------------------------------------
+  # returns an RGB color value as a hex value
+  #-----------------------------------------------------------------------------
+  def to_hex
+  	r = sprintf("%02X", self.red)
+  	g = sprintf("%02X", self.green)
+  	b = sprintf("%02X", self.blue)
+  	return ("#" + r + g + b).upcase
+  end
+  #-----------------------------------------------------------------------------
+  # returns Hex color value as RGB
+  #-----------------------------------------------------------------------------
+  def to_rgb(hex)
```

Comment:
This method doesn't involve `self` at all, so should it be `def self.to_rgb(hex)` instead?

Reply:
Yeah, should definitely be a class method, not like this.